css/how-we-write-css.md

```diff
@@ -0,0 +1,172 @@
+
+#### Utilities
+Utilities are high-specificity, very explicit classes that apply a single rule or a single piece of functionality. They are used as overrides, helper classes, and to quickly build up simple page elements.
+
+Utilities can be used by applying the class to an element, or by including the mixin within the other levels. They should be able to apply to any element and not be bound to any particular markup, and also be independent of any cosmetic styling or branding.
+
+### Folder naming
+
+It can be useful to use numbers at the start of your folder names so that the order of levels is preserved in your text editor. This is optional. Use numbers that allow for the easy insertion of new levels if needed e.g.
+
+```
+10-settings
+20-functions
+30-mixins
+40-base
+50-components
+60-patterns
+70-utilities
+```
+
+### Wider architecture
+
+The way that CSS is written, and particularly the use of **levels**, is designed to fit into a wider componentised architecture. We aim to create shared components that can be used across products and across brands. These components can contain one or more of CSS, Javascript, and templates.
+
+We can think of this architecture as comprising of three distinct groups - company, brand, product. **Company** level components are small self-contained snippets that are applicable to _any_ product and can be customisable to work within their context. If we are talking about CSS then an example might be a set of utilities - `float: left` is the same no matter the brand or product. At the **brand** level we have collections of brand specific components - CSS, Javascript and templates. When talking about CSS, this might include a header component that is specific to a particular brand, and applicable to all products within this brand. Components at this level may have dependencies on company level components. At the most specific level we have **products**. These will contain product specific CSS, Javascript and templating, as well as dependencies at the brand and company levels.
+
+Decisions on what components to use should made **at the product level**. An individual product can decide to use certain company wide components, as well as pick and choose from brand level components (or include them all). This is where the levels are important as relates to CSS - we should be able to import from the company and brand groups at each of the levels, and combine the CSS with our own product CSS at each level, while knowing that specificity is controlled.
+
+## Rules
+
+We implement the following rules within each level:
+
+#### 10 Settings
+* Contains _only_ variable definitions
+
+#### 20 Functions
+* Contains _only_ function definitions
+
+#### 30 Mixins
+* Can include _private_ functions and variables (exist at this level)
+
+#### 40 Base
+* Should avoid type selectors where possible 
```

Comment:
I thought _base_ was primarily aimed at type selectors?
e.g. where you'd find your `html, body, p, ul, ol, a` etc defaults.